consensus/validation_test.go

```diff
@@ -473,6 +473,22 @@ func TestValidateV2Block(t *testing.T) {
 			txn.FileContractRevisions[i].Revision.RenterSignature = renterPrivateKey.SignHash(cs.ContractSigHash(txn.FileContractRevisions[i].Revision))
 			txn.FileContractRevisions[i].Revision.HostSignature = hostPrivateKey.SignHash(cs.ContractSigHash(txn.FileContractRevisions[i].Revision))
 		}
+		for i := range txn.FileContractResolutions {
+			switch r := txn.FileContractResolutions[i].Resolution.(type) {
+			case *types.V2FileContractRenewal:
+				r.RenterSignature = renterPrivateKey.SignHash(cs.RenewalSigHash(*r))
+				r.HostSignature = hostPrivateKey.SignHash(cs.RenewalSigHash(*r))
+				txn.FileContractResolutions[i].Resolution = r
+				break
+			case *types.V2FileContractFinalization:
+				fc := types.V2FileContract(*r)
+				fc.RenterSignature = renterPrivateKey.SignHash(cs.ContractSigHash(fc))
+				fc.HostSignature = hostPrivateKey.SignHash(cs.ContractSigHash(fc))
+
+				txn.FileContractResolutions[i].Resolution = (*types.V2FileContractFinalization)(&fc)
```

Comment:
since `r` is a pointer, you don't need to set `txn.FileContractResolutions[i].Resolution = r`. For the `types.V2FileContractFinalization` branch, I would do:
```go
sigHash := cs.ContractSigHash(types.V2FileContract(*r))
r.RenterSignature = renterPrivateKey.SignHash(sigHash)
r.HostSignature = hostPrivateKey.SignHash(sigHash)
```